ldpd/notification.c

```diff
@@ -243,21 +281,57 @@ recv_notification(struct nbr *nbr, char *buf, uint16_t len)
 		 * initialization, it SHOULD transmit a Shutdown message and
 		 * then close the transport connection".
 		 */
-		if (nbr->state != NBR_STA_OPER && nm.status_code == S_SHUTDOWN)
+		if (nbr->state != NBR_STA_OPER &&
+		    nm.status_code == S_SHUTDOWN) {
+			leconf->stats.session_attempts++;
 			send_notification(nbr->tcp, S_SHUTDOWN,
 			    msg.id, msg.type);
+		}
 
+		leconf->stats.shutdown_rcv_notify++;
 		nbr_fsm(nbr, NBR_EVT_CLOSE_SESSION);
 		return (-1);
 	}
 
-	/* lde needs to know about a few notification messages */
+	/* lde needs to know about a few notification messages
+	 * and update SNMP session counters
+	 */
 	switch (nm.status_code) {
 	case S_PW_STATUS:
 	case S_ENDOFLIB:
 		ldpe_imsg_compose_lde(IMSG_NOTIFICATION, nbr->peerid, 0,
 		    &nm, sizeof(nm));
 		break;
+	case S_NO_HELLO:
+		leconf->stats.session_rejects_hello++;
+		break;
+	case S_PARM_ADV_MODE:
```

Comment:
the same for statistics; i dont think it costs a lot to separte statistics on a separate commit.
this will bring clarity when looking at reviews, if someone wants to add a show vty command refering those stats too.

Reply:
@pguibert6WIND 
Just FYI, I created a PR for LDP SNMP without statistics.  I will mark this PR 7944 as a draft.

https://github.com/FRRouting/frr/pull/8147